projects/core/src/cart/event/cart-event.builder.ts

```diff
@@ -78,10 +78,17 @@ export class CartEventBuilder {
    */
   protected registerMapped<T>(mapping: ActionToEventMapping<T>): () => void {
     const eventStream$ = this.getAction(mapping.action).pipe(
-      withLatestFrom(
-        this.activeCartService.getActive(),
-        this.activeCartService.getActiveCartId()
-      ),
+      switchMap((action) => {
+        // SwitchMap was used instead of withLatestFrom, because we only want to subscribe to cart stream when action is dispatched.
+        // Using withLatestFrom would trigger subscription to cart observables on event subscription and that causes side effects,
+        // such as loading cart when we don't yet need it.
+        return of(action).pipe(
+          withLatestFrom(
+            this.activeCartService.getActive(),
```

Comment:
Just a thought: we don't want to cover this first example case with test? ;-)

Reply:
Added ;)